web/src/components/dashboard/QuickActionTile.test.tsx

```diff
@@ -0,0 +1,75 @@
+import { QuickActionTile } from "@/components/dashboard/QuickActionTile";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { generateQuickActionLink, generateQuickActionTask } from "@/test/factories";
+import { mockPush, useMockRouter } from "@/test/mock/mockRouter";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+function setupMockAnalytics(): typeof analytics {
+  return {
+    ...jest.requireActual("@/lib/utils/analytics").default,
+    event: {
+      ...jest.requireActual("@/lib/utils/analytics").default.event,
+      quick_action_button: {
+        click: {
+          go_to_quick_action_screen: jest.fn(),
+        },
+      },
+    },
+  };
+}
+
+jest.mock("@/lib/utils/analytics", () => setupMockAnalytics());
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+
+const mockAnalytics = analytics as jest.Mocked<typeof analytics>;
+
+describe("<QuickActionTile />", () => {
+  describe("as an external link", () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it("routes to actions/url and triggers analytics when clicked", () => {
+      render(
+        <QuickActionTile
+          type="link"
+          quickAction={generateQuickActionLink({
+            name: "some name",
+            externalRoute: "some-url",
+            filename: "some-filename",
+          })}
+        />
+      );
+      fireEvent.click(screen.getByText("some name"));
+      expect(mockAnalytics.event.quick_action_button.click.go_to_quick_action_screen).toHaveBeenCalledWith(
```

Comment:
dust - should there be an assertion validating the routing doesn't fire?

Reply:
included